sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/models/ResponseBodyHandling.java

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.generic.core.http.models;
+
+import com.generic.core.annotation.ServiceInterface;
+import com.generic.core.http.Response;
+
+/**
+ * Enum that defines how to handle the body of an HTTP response.
+ */
+public enum ResponseBodyHandling {
+    /**
+     * Indicates that the body of an HTTP response shall be entirely ignored. No attempt to read the body will be made
+     * and the connection to the service will be closed as soon as a body-less {@link Response} instance is created.
+     */
+    IGNORE,
+
+    /**
+     * Indicates that the body of an HTTP response shall not be buffered into memory. The connection to the service will
+     * be kept alive to read the body at a future point in time when needed by the created {@link Response} instance.
+     */
+    STREAM,
+
+    /**
+     * Indicates that the body of an HTTP response shall be buffered into memory. The connection to the service will
+     * be closed after a {@link Response} instance containing the body is created. <b>This is the default behavior</b>.
```

Comment:
nit: Per the meeting, this won't always be the default behavior.

We should add class-level documentation that goes into detail on which behavior is the default and when. Ex, `IGNORE` is used when sending a `HEAD` request as per the RFC the body should be ignored, if there is one, in the response, `STREAM` is used when the response payload is `octet-stream`, and possible other response formats, and `BUFFER` is used in almost all other cases.

Reply:
When you talk about class-level documentation, what classes are you referring to?

Reply:
`STREAM` is currently used only for operations that return an `InputStream`. Should we stick to that or base it on Content-Type? If that were the case, which ones do you suggest we use it with?

Reply:
From our offline discussion, we'll use `STREAM` for `application/octet-stream` responses.